src/store/Store.js

```diff
@@ -68,6 +78,26 @@ class Store {
   }
 
   /**
+   * Replaces the state for only the keys in the updated state. Notifies all listeners of state change.
+   * @param {object} state the new (partial) redux state
+   */
+  patchState(difference) {
+    const state = Object.assign({}, this.state);
+
+    difference.forEach(({change, key, value}) => {
+      if (change === 'updated') {
```

Comment:
Love this. I like what you did before with the switch/case ladder - mind doing that here?